libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFUserAccountManager.m

```diff
@@ -589,29 +601,49 @@ - (BOOL)hasCommunityChanged {
     }
 }
 
-- (void)userChanged:(SFUserAccountChange)change {
-    if (![self.lastChangedOrgId isEqualToString:self.currentUser.credentials.organizationId]) {
-        self.lastChangedOrgId = self.currentUser.credentials.organizationId;
+- (void)userChanged:(SFUserAccount *)user change:(SFUserAccountChange)change {
+    [self notifyUserChange:SFUserAccountManagerDidChangeUserDataNotification withUser:user andChange:change];
+}
+
+- (void)notifyCurrentUserChange:(SFUserAccountChange)change {
+    SFUserAccount *user  = [self currentUser];
+    if (![self.lastChangedOrgId isEqualToString:user.credentials.organizationId]) {
+        self.lastChangedOrgId = user.credentials.organizationId;
         change |= SFUserAccountChangeOrgId;
         change &= ~SFUserAccountChangeUnknown; // clear the unknown bit
     }
 
     if (![self.lastChangedUserId isEqualToString:self.currentUser.credentials.userId]) {
-        self.lastChangedUserId = self.currentUser.credentials.userId;
+        self.lastChangedUserId = user.credentials.userId;
```

Comment:
Plus, it would be super sweet to get rid of all of these disconnected state management variables that get out of sync at the drop of a hat.